### PR TITLE
Added methods to read and delete app/user id

### PR DESCRIPTION
**Pull request by blarghmatey**
Added methods and tests for reading and deleting from the app-id and user-id endpoints

### CODE REVIEW

File: hvac/tests/test_integration.py [hunk 1]
```diff
@@ -290,20 +296,6 @@ def test_invalid_token(self):
         client = create_client(token='not-a-real-token')
         assert not client.is_authenticated()
 
-    def test_illegal_token(self):
```
Comment (ianunruh) [inline hvac/tests/test_integration.py]: Why did you remove these 2 tests?

Comment (blarghmatey) [inline hvac/tests/test_integration.py]: That must have been an accidental delete. I'll restore them and update the PR.

**Comment by ianunruh** [comment]
Thank you!
